src/broadcast/listenbrainzjsonfactory.cpp

```diff
@@ -0,0 +1,38 @@
+
+#include <QDateTime>
+#include <QJsonArray>
+#include <QJsonDocument>
+#include <QJsonObject>
+
+#include "listenbrainzjsonfactory.h"
+
+
+QByteArray ListenBrainzJSONFactory::getJSONFromTrack(TrackPointer pTrack, JsonType type) {
+    QJsonObject jsonObject;
+    QString stringType;
+    if (type == NowListening) {
+        stringType = "playing_now";
+    }
+    else {
+        stringType = "single";
+    }
+
+    QJsonArray payloadArray;
+    QJsonObject payloadObject;
+    QJsonObject metadataObject;
+    QString title = pTrack->getTitle();
+    QString artist = pTrack->getArtist();
+    metadataObject.insert("artist_name",artist);
+    metadataObject.insert("track_name",title);
+    payloadObject.insert("track_metadata",metadataObject);
+    qint64 timeStamp = QDateTime::currentSecsSinceEpoch();
```

Comment:
This is only available from QT 5.8 I am on Qt 5.2

Reply:
Can we backport this from the Qt source?

Reply:
Hmmm I think trying to find a solution for your version will be better, if I can't I'll do what you said. Also, which version is master supposed to compile in?